### PR TITLE
Strict and non-strict transitive property names

### LINKED ISSUE #2087: Give consistent names to transitivity between strict and non-strict orders
e.g. sometimes in `Data.Integer.Properties` its `≤-<-trans` and in `Data.Nat.Properties` its `<-transʳ`.

### PR DESCRIPTION
**Pull request by jamesmckinna**
Tackles #2087 (and bits of #2088): the names have... got a bit weird/unwieldy, but I think are (jointly) consistent now. And maybe they are simplifiable to remove the superscripts, except that they tee up what the *result* relation should be, which is otherwise opaque from the name, you have to look at the types to disambiguate... :-(

### CODE REVIEW

File: src/Data/Nat/DivMod.agda [hunk 1]
```diff
@@ -119,12 +123,12 @@ m%n≤n m n = <⇒≤ (m%n<n m n)
...
 m%n≤m m (suc n-1) = a[modₕ]n≤a 0 m n-1
 
-m≤n⇒m%n≡m : ∀ {m n} → m ≤ n → m % suc n ≡ m
+m≤n⇒m%n≡m : m ≤ n → m % suc n ≡ m
 m≤n⇒m%n≡m {m} {n} m≤n with ≤⇒≤″ m≤n
 ... | less-than-or-equal {k} refl = a≤n⇒a[modₕ]n≡a 0 (m + k) m k
 
 m<n⇒m%n≡m : ∀ {m n} .⦃ _ : NonZero n ⦄ → m < n → m % n ≡ m
-m<n⇒m%n≡m {m} {suc n} m<n = m≤n⇒m%n≡m (<⇒≤pred m<n)
+m<n⇒m%n≡m {m} {n = suc _} m<n = m≤n⇒m%n≡m (<⇒≤pred m<n)
```
Comment (gallais) [inline src/Data/Nat/DivMod.agda:131]:
```suggestion
m<n⇒m%n≡m {n = suc _} m<n = m≤n⇒m%n≡m (<⇒≤pred m<n)
```

File: src/Data/Nat/DivMod.agda [hunk 2]
```diff
@@ -119,12 +123,12 @@ m%n≤n m n = <⇒≤ (m%n<n m n)
 m%n≤m : ∀ m n .{{_ : NonZero n}} → m % n ≤ m
 m%n≤m m (suc n-1) = a[modₕ]n≤a 0 m n-1
 
-m≤n⇒m%n≡m : ∀ {m n} → m ≤ n → m % suc n ≡ m
+m≤n⇒m%n≡m : m ≤ n → m % suc n ≡ m
 m≤n⇒m%n≡m {m} {n} m≤n with ≤⇒≤″ m≤n
 ... | less-than-or-equal {k} refl = a≤n⇒a[modₕ]n≡a 0 (m + k) m k
 
 m<n⇒m%n≡m : ∀ {m n} .⦃ _ : NonZero n ⦄ → m < n → m % n ≡ m
```
Comment (gallais) [inline src/Data/Nat/DivMod.agda:130]:
```suggestion
m<n⇒m%n≡m : .⦃ _ : NonZero n ⦄ → m < n → m % n ≡ m
```

File: src/Data/Nat/DivMod.agda [hunk 3]
```diff
@@ -182,11 +186,11 @@ m<[1+n%d]⇒m≤[n%d] {m} n (suc d-1) = k<1+a[modₕ]n⇒k≤a[modₕ]n 0 m n d-
 ------------------------------------------------------------------------
 -- Properties of _/_
 
-/-congˡ : ∀ {m n o : ℕ} .{{_ : NonZero o}} →
+/-congˡ :  .{{_ : NonZero o}} →
           m ≡ n → m / o ≡ n / o
```
Comment (gallais) [inline src/Data/Nat/DivMod.agda:190]:
```suggestion
/-congˡ : .{{_ : NonZero o}} → m ≡ n → m / o ≡ n / o
```

**Review by gallais — COMMENTED** [review]
Nice!

File: src/Data/Nat/DivMod.agda (hunk 3 above)
Comment (jamesmckinna) [reply src/Data/Nat/DivMod.agda:190]: And the subsequent lemma, for uniformity of layout.

**Comment by jamesmckinna** [comment]
Merge conflict resolved.

**Comment by MatthewDaggitt** [comment]
> And maybe they are simplifiable to remove the superscripts, except that they tee up what the result relation should be, which is otherwise opaque from the name, you have to look at the types to disambiguate... :-(

I would be strongly in favour of removing the superscripts. The resulting relation is _always_ going to the strict version as the non-strict version can be obtained from a simple cast from strict to non-strict.

Once, that's done, looks good to me!

**Review by MatthewDaggitt — COMMENTED** [review]
Looks good.

**Pull request merged**